src/uct/ib/mlx5/exp/ib_exp_md.c

```diff
@@ -309,6 +273,123 @@ static ucs_status_t uct_ib_mlx5_exp_reg_atomic_key(uct_ib_md_t *ibmd,
     }
 err_free_umr:
     UCS_PROFILE_CALL(ibv_dereg_mr, umr);
+err:
+    return status;
+}
+#endif
+
+ucs_status_t uct_ib_mlx5_exp_reg_ksm(uct_ib_mlx5_md_t *md,
```

Comment:
is it possible that instead of exposing both reg_atomic_key and reg_multithreaded, we would expose something like "reg_indirect_mr" which would be used for both cases? so we could move more code from exp/devx/mlx5 to base and avoid duplication.